oak_functions/loader/src/logger.rs

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chrono::{SecondsFormat, Utc};
+use log::{Level, LevelFilter};
+use std::default::Default;
+
+/// A simple logger that splits logging between writing logs that contain only public, non-sensitive
+/// content and writing logs that could potentially contain sensitive content.
+///
+/// Writing of potentially sensitive content will be ignored unless the "oak-unsafe" feature is
+/// enabled.
+pub(crate) struct Logger {
+    max_level: LevelFilter,
+}
+
+impl Logger {
+    /// Creates a new logger with the specified maximum `LevelFilter`.
+    #[allow(dead_code)]
+    pub(crate) fn new(max_level: LevelFilter) -> Self {
+        Self { max_level }
+    }
+
+    /// Logs the message at the specified `Level`, but only if the "oak-unsafe" feature is enabled.
+    #[allow(dead_code)]
+    #[allow(unused_variables)]
+    pub(crate) fn log_unsafe(self: &Self, level: Level, message: &str) {
+        #[cfg(feature = "oak-unsafe")]
+        {
+            self.log(level, message);
+        }
+    }
+
+    /// Logs a message that contains only public, non-sensitive content at the specified `Level`.
+    ///
+    /// All code that uses this function should be inspected to ensure that the message can never
+    /// contain any sensitive or non-public information.
+    #[allow(dead_code)]
+    pub(crate) fn log_public(self: &Self, level: Level, message: &str) {
```

Comment:
Not very actionable, but it seems a bit odd that, if we were annotating these functions with `unsafe`, actually `log_public` would be `unsafe` and `log_unsafe` would not be. I think it's fine as it is, but it makes me think whether we should rethink the terminology or the API.

Reply:
Perhaps rename `log_unsafe` to `log_sensitive`?

Reply:
Done. Also renamed feature to "oak-insecure" seeing that "unsafe" has a specific meaning in Rust.

Reply:
I undid the rename of the feature, as it was referenced in the runner etc. Could be done in future in a separate PR.